base_layer/key_manager/src/key_manager.rs

```diff
@@ -84,13 +87,17 @@ where
 
     /// Derive a new private key from master key: derived_key=SHA256(master_key||branch_seed||index)
     pub fn derive_key(&self, key_index: u64) -> Result<DerivedKey<K>, ByteArrayError> {
-        let concatenated = format!(
-            "{}{}{}",
-            self.seed.entropy().to_vec().to_hex(),
-            self.branch_seed,
-            key_index
-        );
-        match K::from_bytes(D::digest(&concatenated.into_bytes()).as_slice()) {
+        // apply domain separation to generate derive key. Under the hood, the hashing api prepends the length of each
+        // piece of data for concatenation, reducing the risk of collisions due to redundance of variable length
+        // input
+        let derive_key = DomainSeparatedHasher::<D, GenericHashDomain>::new(DOMAIN_SEPARATION_LABEL)
+            .chain(self.seed.entropy().to_vec().to_hex())
```

Comment:
Any reason to use a hex encoding here? It seems unnecessary, and the use of the hashing API is already a breaking change.

Reply:
You are totally right Aaron, one can simply chain the seed entropy as it is.